### PR TITLE
Fix bops calculation

### PR DESCRIPTION
**Pull request by irenaby**
## Pull Request Description:
Support for bops computation on virtual nodes in RU calculator.
Updated virtual graph substitutions to fix activation & weights RU calculation on virtual graph. 
Use RU calculator for bops in mixed precision search.
Added integration tests for RU calculator (framework -> graph preparation -> RU calculator, original vs virtual graph).
Added verification that target RU is satisfied by final RU in MP tests.

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

**Comment by ofirgo** [comment]
@irenaby 
I don't remember from all we discussed whether this PR changes any behavior that is not BOPs only related (activation/weights memory estimation, any RU computation in MP or RU data besides BOPs).
If there is something like that, please specify this in the PR description.

File: model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py [hunk 1]
```diff
@@ -473,26 +505,56 @@ def compute_node_bops(self,
...
+        node_mac = self.fw_impl.get_node_mac_operations(orig_w_node, self.fw_info)
+        if node_mac == 0:
+            return node_mac
+
+        # find the activation node from which to get quantization info and for which to look in custom configuration
+        if isinstance(n, VirtualActivationWeightsNode):
+            # we don't need the original node (and cannot use it for custom configuration anyway)
+            a_node = n
+        else:
+            a_node = get_input_activation_if_composable(self.graph, n, warn=False)
```
Comment (elad-c) [inline model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py:547]: In what case will we get here? shouldn't activation nodes followed by kernel nodes be already fused into `VirtualActivationWeightsNode`

File: tests_pytest/keras_tests/integration_tests/core/mixed_precision/test_resource_utilization.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        inputs = Input(shape=self.bhwc_input_shape[1:])
+        x = Conv2D(filters=8, kernel_size=5)(inputs)
+        x = tf.add(x, np.ones((14, 14, 8)))  # => activation with const in the composed node
+        x = DepthwiseConv2D(kernel_size=3, depth_multiplier=2)(x)  # => Virtual activation in the composed node
+        x = Conv2DTranspose(filters=12, kernel_size=5)(x)
+        x = Flatten()(x)
+        outputs = Dense(10)(x)
+        return keras.Model(inputs=inputs, outputs=outputs)
+
+    def _build_mult_output_activation_model(self):
```
Comment (elad-c) [inline tests_pytest/keras_tests/integration_tests/core/mixed_precision/test_resource_utilization.py:45]: mult input?

File: tests_pytest/pytorch_tests/integration_tests/core/mixed_precision/test_resource_utilization.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                x = x + to_torch_tensor(torch.ones((8, 14, 14)))
+                x = self.dw_conv(x)
+                x = self.conv_tr(x)
+                x = self.flatten(x)
+                x = self.linear(x)
+                return x
+
+        return Model()
+
+    def _build_mult_output_activation_model(self):
```
Comment (elad-c) [inline tests_pytest/pytorch_tests/integration_tests/core/mixed_precision/test_resource_utilization.py:55]: mult input?

**Review by elad-c — APPROVED** [review]

File: model_compression_toolkit/core/common/graph/virtual_activation_weights_node.py [hunk 4]
```diff
@@ -126,61 +128,57 @@ class VirtualActivationWeightsNode(BaseNode):
...
+                         reuse=weights_node.reuse,
+                         reuse_group=weights_node.reuse_group,
+                         quantization_attr=weights_node.quantization_attr,
+                         has_activation=False)
 
         self.original_activation_node = act_node
         self.original_weights_node = weights_node
 
         v_candidates = []
+        kernel_attr = fw_info.get_kernel_op_attributes(weights_node.type)[0]
```
[the remark this answers is not in the document]
Comment (irenaby) [reply model_compression_toolkit/core/common/graph/virtual_activation_weights_node.py:168]: Not supported, added check that no other atrrs are configurable.

File: model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py (hunk 1 above)
Comment (irenaby) [reply model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py:547]: In case we run on the original graph,  added comment.

File: tests_pytest/keras_tests/integration_tests/core/mixed_precision/test_resource_utilization.py (hunk 2 above)
Comment (irenaby) [reply tests_pytest/keras_tests/integration_tests/core/mixed_precision/test_resource_utilization.py:45]: its input activation with mult output
